src/Microsoft.SqlTools.ServiceLayer/Profiler/ProfilerService.cs

```diff
@@ -170,48 +171,57 @@ public void InitializeService(ServiceHost serviceHost)
-        internal ProfilerSession StartSession(string sessionId, ConnectionInfo connInfo)
+        internal async Task HandlePauseProfilingRequest(PauseProfilingParams parameters, RequestContext<PauseProfilingResult> requestContext)
         {
-            // create a new XEvent session and Profiler session
-            var xeSession = this.XEventSessionFactory.CreateXEventSession(connInfo);
-            var profilerSession = new ProfilerSession()
+            try
             {
-                SessionId = sessionId,
-                XEventSession = xeSession
-            };
-
-            // start monitoring the profiler session
-            monitor.StartMonitoringSession(profilerSession);
+                monitor.PauseViewer(parameters.OwnerUri);
 
-            return profilerSession;
+                await requestContext.SendResult(new PauseProfilingResult
+                {
+                    Succeeded = true
+                });
+            }
+            catch (Exception e)
+            {
+                await requestContext.SendError(e);
+            }
         }
 
         /// <summary>
-        /// Create a new XEvent sessions per the IXEventSessionFactory contract
+        /// Starts a new profiler session or connects to an existing session
+        /// for the provided connection and template info
         /// </summary>
-        public IXEventSession CreateXEventSession(ConnectionInfo connInfo)
+        /// <returns>
+        /// The XEvent Session ID that was started
+        /// </returns>
+        internal int StartSession(string ownerUri, string template, ConnectionInfo connInfo)
         {
-            var sqlConnection = ConnectionService.OpenSqlConnection(connInfo);
-            SqlStoreConnection connection = new SqlStoreConnection(sqlConnection);
-            Session session = ProfilerService.GetOrCreateSession(connection, "Profiler");
+            // create a new XEvent session and Profiler session
+            var xeSession = this.XEventSessionFactory.GetOrCreateXEventSession(template, connInfo);
 
-            // create xevent session wrapper
-            return new XEventSession()
-            {
-                Session = session
-            };
+            // start monitoring the profiler session
+            monitor.StartMonitoringSession(ownerUri, xeSession);
+
+            return xeSession.ID;
         }
 
         /// <summary>
-        /// Gets an existing XEvent session or creates one if no matching session exists.
+        /// Gets or creates an XEvent session with the given template per the IXEventSessionFactory contract
         /// Also starts the session if it isn't currently running
         /// </summary>
-        private static Session GetOrCreateSession(SqlStoreConnection connection, string sessionName)
+        public IXEventSession GetOrCreateXEventSession(string template, ConnectionInfo connInfo)
         {
+            // TODO: Change this to handle different names based off of templates
```

Comment:
It's helpful to link a TODO to a specific GitHub issue, if you have one tracking this work. The top-level profiler issue should be fine